var/spack/repos/builtin/packages/grads/package.py

```diff
@@ -63,10 +63,13 @@ def url_for_version(self, version):
             return url.format(version.up_to(2), version)
 
     # Name of grib2 C library has changed in recent versions
-    with when("+grib2"):
-
-        def patch(self):
+    def patch(self):
+        if self.spec.satisfies("@:2.2.2"):
+            filter_file("png15", "png", "configure")
+        
+        if self.spec.satisfies("+grib2"):
```

Comment:
Looking through the g2c repo, it seems that this constant was changed from `G2_VERSION` to `G2C_VERSION` in g2c@1.8.0. I'd suggest changing this satisfies to:

```
if self.spec.satisfies("^g2c@1.8.0:"):
```